### DISCUSSION TITLE
How to locate control's axaml code through preview like WPF?

### DISCUSSION THREAD

**Discussion by KailangHuo** [AvaloniaUI/Avalonia#13956] [Q&A] [answered]
Hello guys, I'm a rookie in Avalonia. Primarily I use JetBrains rider to develop my Avalonia application, I was wondering if it is applicable to locate a control's axaml code through the preview window just like in WPF development, in which a simple click would do. Or highlight the control while i'm having my related code selected. 
Any help is appreciated!

**Comment by thevortexcloud** [comment]
I don't _think_ it can do this. Someone may correct though. But unlike the WPF designer the Avalonia previewer works by running your actual app. As such all the XAML has been compiled at that point.

Instead Avalonia has runtime tools for things like this. Specifically it has the dev tools which can be opened by pressing F12 while your app is running (assuming you have the package installed).

https://docs.avaloniaui.net/docs/guides/implementation-guides/developer-tools

**Comment by timunie** [comment, marked as answer]
@KailangHuo we want to implement something like that but it's not as easy as in WPF for us. WPF support is added to VS by Microsoft so they have full control of the IDE and much more resources. So for now the previewer is a previewer not more, not less.

**Reply by KailangHuo** [reply]
Thanks for your explanation. At least I don't need to keep searching on this.
